File: ctu_marketplace_new-feature-introduction_footer/ctu-marketplace-web/src/services/faq.service.js
import api from '../api/faq';
import { FAQ_URL } from '../environments/constraints';

class FaqService {
  
    getAll() {
        return api.get(FAQ_URL);
    }
    
    // getById(id) {
    //     return api.get(`${FAQ_URL}/${id}`);
    // }

    // create(data) {
    //     return api.post(FAQ_ADMIN_URL, data);
    // }

    // update(id, data) {
    //     return api.put(`${FAQ_ADMIN_URL}/${id}`, data);
    // }

    // delete(id) {
    //     return api.delete(`${FAQ_ADMIN_DELETE_URL}/${id}`);
    // }

}

export default new FaqService();

File: ctu_marketplace_new-feature-introduction_footer/ctu-marketplace-v2/src/main/java/com/ctu/marketplace/service/FaqService.java
package com.ctu.marketplace.service;

import java.util.List;

import com.ctu.marketplace.dto.request.FaqDto;
import com.ctu.marketplace.entity.Faq;

public interface FaqService {
    List<Faq> getAll();

    Faq getById(Long faqId);

    Faq create(FaqDto faqDto);

    Faq update(Long faqId, FaqDto faqDto);

    boolean deleteById(Long faqId);
}
